Add MirrorMaker2 topic details to external integration howto [HH-2946] (#2242)
aiven/devportal: docs/products/kafka/kafka-mirrormaker/howto/integrate-external-kafka-cluster.rst

5. The external Apache Kafka cluster is now available under the alias defined in the **Endpoint name** parameter

Permissions and configuration when using external integration
-------------------------------------------------------------

> By default Apache Kafka® MirrorMaker2 creates topics for Apache Kafka® MirrorMaker2 internal data and writes messages to those topics. When using external integration to Kafka the integration data defines the connecting user. For Apache Kafka® MirrorMaker2 to be able to create and write the connecting user needs permissions to do so. It is a requirement to add ACLs to the corresponding Kafka cluster for the connecting users.

Author: Done.

> By default Apache Kafka® MirrorMaker2 creates topics for Apache Kafka® MirrorMaker2 internal data and writes messages to those topics. When using external integration to Kafka the integration data defines the connecting user. For Apache Kafka® MirrorMaker2 to be able to create topics and write into them the connecting user needs permissions to do so. It is a requirement to add ACLs to the corresponding Kafka cluster for the connecting users.

Reviewer: ```suggestion
By default, Apache Kafka® MirrorMaker 2 creates topics for its internal data and writes messages to these topics. When integrating externally with Kafka, the integration data defines the connecting user. For Apache Kafka® MirrorMaker 2 to create topics and write to them, the connecting user must have the necessary permissions. Adding ACLs to the corresponding Kafka cluster for the connecting users is required.
```

Author: Done.

For external source the read permission is required to the mirrored topics and on external target the write permission is required to the mirrored topics.

> For the internal topics a read and write permissions is required for connecting user.

Author: Done.

> For external source the read permission is required to the mirrored topics and on external target the write permission is required to the mirrored topics.
> For the internal topics the read and write permissions are required for the connecting user.

Reviewer: ```suggestion
To mirror topics, the user connecting to external sources must have read permission for the mirrored topics. Similarly, for external targets, the connecting user needs write permission for the mirrored topics. 
Additionally, the connecting user must have both read and write permissions for the internal topics.
```

Author: Done.

> The set of topics created by Apache Kafka® MirrorMaker2 to source cluster.

Author: Done.

> The set of topics created by Apache Kafka® MirrorMaker2 to source cluster is the following.

Author: Done.

.. list-table::
  :header-rows: 1

  * - Topic
    - | Replication factor
      | (default value)
    - | Partitions
      | (default value)
    - | Cleanup policy
      | (required policy)
  * - ``__consumer_offsets``
    - 3
    - 50
    - compact

>   * - ``heartbeats``

Reviewer 2: I assume, it also replicates `<target cluster alias>.heartbeats`?

Author: I'll fire up the env again to check this. I looked the topics that get created to source and target and I don't think I saw this one. But this could be an artifact of how to configure MM2.

Author: This topic is created only when heartbeat emit is enabled. I'll add this detail.

Author: Hmm..no. The `<source cluster alias>.heartbeats` is the only created and mirrored heartbeat topic.

    - 3
    - 1
    - compact
  * - ``mm2-configs.<target cluster alias>.internal``
    - 3
    - 1
    - compact
  * - | ``mm2-offset-syncs.<target cluster alias>.internal``
      | see `Offsets sync topic location`_
    - 3
    - 1
    - compact
  * - ``mm2-offsets.<target cluster alias>.internal``
    - 3
    - 25
    - compact
  * - ``mm2-status.<target cluster alias>.internal``
    - 3
    - 5
    - compact

> The set of topics created by Apache Kafka® MirrorMaker2 to target cluster.

Author: Done.

> The set of topics created by Apache Kafka® MirrorMaker2 to target cluster is the following:

Reviewer: ```suggestion
Apache Kafka® MirrorMaker2 creates the following set of topics in the target cluster:
```

Author: Done.

.. list-table::
  :header-rows: 1

  * - Topic
    - | Replication factor
      | (default value)
    - | Partitions
      | (default value)
    - | Cleanup policy
      | (required policy)
  * - ``__consumer_offsets``
    - 3
    - 50
    - compact
  * - ``heartbeats``
    - 3
    - 1
    - compact
  * - ``<source cluster alias>.checkpoints.internal``
    - 3
    - 1
    - compact
  * - ``<source cluster alias>.heartbeats``
    - 3
    - 1
    - compact
  * - ``mm2-configs.<source cluster alias>.internal``
    - 3
    - 1
    - compact
  * - ``mm2-offsets.<source cluster alias>.internal``
    - 3
    - 25
    - compact
  * - ``mm2-status.<source cluster alias>.internal``
    - 3
    - 5
    - compact

> If using heartbeat emitting to source cluster, configuration is called ``emit_backward_heartbeats_enabled``, the read and write is required to ``mm2-offsets.<target cluster alias>.internal`` topic at the external source cluster. When disabled the topic and permissions are not required at the source cluster.

Reviewer: ```suggestion
If using heartbeat emitting to the source cluster (configuration: ``emit_backward_heartbeats_enabled``), the connecting user needs read and write access to the ``mm2-offsets.<target cluster alias>.internal`` topic at the external source cluster. If disabled, this topic and permissions are not required at the source cluster. 
```

Author: Done.

> If using heartbeat emitting to source cluster, configuration is called ``emit_backward_heartbeats_enabled``, the read and write is required to ``mm2-offsets.<source cluster alias>.internal`` topic at the external source cluster. When disabled the topic and permissions are not required at the source cluster.
> Similarly if using heartbeat emitting to target cluster, configuration is called ``emit_heartbeats_enabled``, the read and write is required to ``mm2-offsets.<target cluster alias>.internal``. When disabled the topic and permissions are not required at the target cluster.

Reviewer 2: IIUC, it should be
```suggestion
If using heartbeat emitting to source cluster, configuration is called ``emit_backward_heartbeats_enabled``, the read and write is required to ``mm2-offsets.<target cluster alias>.internal`` topic at the external source cluster. When disabled the topic and permissions are not required at the source cluster.
Similarly if using heartbeat emitting to target cluster, configuration is called ``emit_heartbeats_enabled``, the read and write is required to ``mm2-offsets.<source cluster alias>.internal``. When disabled the topic and permissions are not required at the target cluster.
```

Author: Done.

> Similarly if using heartbeat emitting to target cluster, configuration is called ``emit_heartbeats_enabled``, the read and write is required to ``mm2-offsets.<source cluster alias>.internal``. When disabled the topic and permissions are not required at the target cluster.

Reviewer: ```suggestion
Similarly, if enabling heartbeat emitting to the target cluster (configuration: ``emit_heartbeats_enabled``), the connecting user requires read and write access to ``mm2-offsets.<source cluster alias>.internal``. If disabled, this topic and permissions are not necessary at the target cluster.
```

Author: Done.

.. note::

>    Summarizing the requirements, configure the ACLs for both the source and target cluster such that the Apache Kafka® MirrorMaker 2 service can describe and create topics, as well as produce and consume messages.

Reviewer 2: Maybe we need to mention that this concerns only integrations with "external" Kafkas

Author: The page is for external kafka clusters, I think highlighting is makes sense.

Author: Done.

>    Summarizing the requirements for external Kafka cluster integration, configure the ACLs for both the source and target cluster such that the Apache Kafka® MirrorMaker 2 service can describe and create topics, as well as produce and consume messages.

Reviewer: ```suggestion
   To summarize, for external Kafka cluster integration, configure the ACLs for both the source and target clusters. This will enable Apache Kafka® MirrorMaker 2 to describe and create topics, as well as produce and consume messages.
```

Author: Done.

Offsets sync topic location
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> Offsets sync topic can be either in source cluster or target cluster based on the configuration value of ``offset-syncs.topic.location``. Default location is ``source``. Connecting user must have permission to read and write this topic.

Reviewer: ```suggestion
Offsets sync topic can be either in the source or target cluster based on the configuration value of ``offset-syncs.topic.location. ``. The default location is ``source``. The connecting user must have permission to read and write this topic.
```

Author: Done.

> If location is configured as ``target`` this topic will be created to the target cluster.

Reviewer: ```suggestion
If the location is configured as ``target, `` this topic will be created in the target cluster.
```

Reviewer: Added some minor suggestions to the content (mostly trying to use active voice as much as possible).

Author: Done.
